### ISSUE TITLE
How to get the same output as the example?

### ISSUE THREAD

**Issue by lngmngx** [RF5/simple-autovc#1]
Hi there. I encountered some errors while running the quickstart code. After modifying something in step 2, I managed to run the code without errors and save the converted utterance file. However, the audio I got was just a bunch of noise. Is there anything that I was not supposed to change in order to get the same result as the example?

Here's what I modified:
In Step 2, I changed
```
    src_embedding = model(sse_src_mel[None].to(device))

    trg_embedding = model(sse_trg_mel[None].to(device))
```
into
```
    src_embedding = sse.model(sse_src_mel[None].to(device))

    trg_embedding = sse.model(sse_trg_mel[None].to(device))
```

and
```
    s_src = sse_src_mel.to(device)
    s_trg = sse_trg_mel.to(device)
```
into
```
    s_src = src_embedding.to(device)
    s_trg = trg_embedding.to(device)
```

**Comment by RF5** [comment]
Hi there, thank you for raising this issue. I just checked through the code and the issue was on my end -- there were some bugs from my local testing that crept into the final commits. 

I have made a few commits which should fix the issues you were having. Please try the quickstart code again and let me know if your problem is resolved (you may need to give the `force_reload=True` argument to the `torch.hub.load` calls to ensure that it does not use an old cached version of the code). 

In detail, what was happening is that by calling `sse.model()` instead of just `sse()`, the spectrograms were not being normalized in the way the speaker embedder expected, causing it to output non-sensical speaker embeddings, thereby making your final converted utterance sound like pure noise.

**Comment by lngmngx** [comment]
> Hi there, thank you for raising this issue. I just checked through the code and the issue was on my end -- there were some bugs from my local testing that crept into the final commits.
> 
> I have made a few commits which should fix the issues you were having. Please try the quickstart code again and let me know if your problem is resolved (you may need to give the `force_reload=True` argument to the `torch.hub.load` calls to ensure that it does not use an old cached version of the code).
> 
> In detail, what was happening is that by calling `sse.model()` instead of just `sse()`, the spectrograms were not being normalized in the way the speaker embedder expected, causing it to output non-sensical speaker embeddings, thereby making your final converted utterance sound like pure noise.

It worked like a charm. Thanks!

Now if I want to use an unseen speaker as the target, should I train all three models on the new data or only the autovc model? (I got good result when I used an unseen speaker as the source, but got noise just like before if I made it the target, so I assume I need to train the model(s) on the corresponding data?)

**Comment by RF5** [comment]
The AutoVC model is a zero-shot model, meaning you can use an unseen speaker as the target speaker without any re-training. Namely, all you have to do is replace the audio file used for `sse_trg_mel` with an audio file from an unseen speaker:

```python
# Get embedding for target speaker
sse_trg_mel = sse.melspec_from_file('<audio file from unseen speaker>')
with torch.no_grad(): 
    trg_embedding = sse(sse_trg_mel[None].to(device))
```

Then using that target speaker embedding to do the conversion will yield a converted utterance. As for conversion quality, this is where AutoVC can struggle with out-of-domain data (e.g. when your target speaker is from a different dataset than the training dataset). So if your unseen target speaker is from e.g. Librispeech, then the AutoVC model here will not do so well because it is trained on VCTK dataset and works best for VCTK (seen or unseen) speakers. To improve performance on your unseen speaker, the optimal thing would be to fine-tune AutoVC on a dataset that includes your desired unseen speaker. 

So, to answer your final question: **you only need to fine-tune / retrain AutoVC on the new data**, the HiFiGAN vocoder and speaker embedding model already work really well on out-of-domain data (but you can fine-tune them as well if you want to squeeze some more performance out of them).

**Comment by lngmngx** [comment]
> The AutoVC model is a zero-shot model, meaning you can use an unseen speaker as the target speaker without any re-training. Namely, all you have to do is replace the audio file used for `sse_trg_mel` with an audio file from an unseen speaker:
> 
> ```python
> # Get embedding for target speaker
> sse_trg_mel = sse.melspec_from_file('<audio file from unseen speaker>')
> with torch.no_grad(): 
>     trg_embedding = sse(sse_trg_mel[None].to(device))
> ```
> 
> Then using that target speaker embedding to do the conversion will yield a converted utterance. As for conversion quality, this is where AutoVC can struggle with out-of-domain data (e.g. when your target speaker is from a different dataset than the training dataset). So if your unseen target speaker is from e.g. Librispeech, then the AutoVC model here will not do so well because it is trained on VCTK dataset and works best for VCTK (seen or unseen) speakers. To improve performance on your unseen speaker, the optimal thing would be to fine-tune AutoVC on a dataset that includes your desired unseen speaker.
> 
> So, to answer your final question: **you only need to fine-tune / retrain AutoVC on the new data**, the HiFiGAN vocoder and speaker embedding model already work really well on out-of-domain data (but you can fine-tune them as well if you want to squeeze some more performance out of them).

Thank you very much! I will try that.
